test/LibraryManager.Test/Providers/Unpkg/UnpkgCatalogTest.cs

```diff
@@ -129,42 +121,69 @@ public async Task GetLibraryCompletionSetAsync_Versions()
         [TestMethod]
         public async Task GetLatestVersion_LatestExist()
         {
-            CancellationToken token = CancellationToken.None;
-            const string libraryId = "bootstrap@3.3.0";
-            string[] existing = libraryId.Split('@');
-            string result = await _catalog.GetLatestVersion(existing[0], false, token);
-
-            // It can return null value.
-            if (result != null)
-            {
-                Assert.AreNotEqual(existing[1], result);
-            }
+            const string libraryName = "fakeLibrary";
+            Mocks.WebRequestHandler handler = new Mocks.WebRequestHandler().SetupVersions("fakeLibrary");
+            UnpkgCatalog sut = SetupCatalog(handler);
+
+            string result = await sut.GetLatestVersion(libraryName, false, CancellationToken.None);
+
+            Assert.AreEqual("1.0.0", result);
         }
 
         [TestMethod]
+        [Ignore] // TODO: GetLatestVersion currently doesn't distinguish stable and pre-release versions
         public async Task GetLatestVersion_PreRelease()
         {
-            CancellationToken token = CancellationToken.None;
-            const string libraryId = "bootstrap@3.3.0";
-            string[] existing = libraryId.Split('@');
-            string result = await _catalog.GetLatestVersion(existing[0], true, token);
-
-            // It can return null value.
-            if (result != null)
-            {
-                Assert.AreNotEqual(existing[1], result);
-            }
+            const string libraryName = "fakeLibrary";
+            Mocks.WebRequestHandler handler = new Mocks.WebRequestHandler().SetupVersions("fakeLibrary");
+            UnpkgCatalog sut = SetupCatalog(handler);
+
+            string result = await sut.GetLatestVersion(libraryName, true, CancellationToken.None);
+
+            Assert.AreEqual("2.0.0-beta", result);
         }
 
         [TestMethod]
         public async Task GetLibraryCompletionSetAsync_WithNullNpmPackageInfoVersions_ReturnsNoCompletions()
         {
-            CompletionSet result = await _catalog.GetLibraryCompletionSetAsync("@", 1);
+            UnpkgCatalog sut = SetupCatalog();
+
+            CompletionSet result = await sut.GetLibraryCompletionSetAsync("@", 1);
 
             Assert.AreEqual(1, result.Start);
             Assert.AreEqual(0, result.Length);
             Assert.AreEqual(0, result.Completions.Count());
             Assert.AreEqual(CompletionSortOrder.Version, result.CompletionType);
         }
     }
+
+    internal static class UnpkgCatalogSetups
```

Comment:
Classes containing extension methods cannot be nested, so it can't be private or else it would be uncallable.

I don't yet see a need for sharing these setups, they are specifically for the requests that this component parses.  If a need to share them does arise, then it would make sense to move them to a central place, and probably refactor or at least rename them because there's similar helpers for each catalog (since they each talk to a different endpoint and get slightly different payloads back).